src/main/java/com/openshift/jenkins/plugins/pipeline/dsl/OpenShiftScaler.java

```diff
@@ -84,55 +94,91 @@ public BuildStepMonitor getRequiredMonitorService() {
-        }
-
-        @Override
-        public Step newInstance(Map<String, Object> arguments) throws Exception {
-            if (!arguments.containsKey("deploymentConfig") && !arguments.containsKey("depCfg"))
-            	throw new IllegalArgumentException("need to specify deploymentConfig");
-            Object depCfg = arguments.get("deploymentConfig");
-            if (depCfg == null || depCfg.toString().length() == 0)
-            	depCfg = arguments.get("depCfg");
-            if (depCfg == null || depCfg.toString().length() == 0)
-            	throw new IllegalArgumentException("need to specify deploymentConfig");
-            if (!arguments.containsKey("replicaCount"))
-            		throw new IllegalArgumentException("need to specif replicaCount");
-            OpenShiftScaler step = new OpenShiftScaler(depCfg.toString(),
-            		arguments.get("replicaCount").toString());
-            
-            if (arguments.containsKey("waitTime")) {
-            	Object waitTime = arguments.get("waitTime");
-            	if (waitTime != null)
-            		step.setWaitTime(waitTime.toString());
-            }
-            if (arguments.containsKey("verifyReplicaCount")) {
-            	Object verifyReplicaCount = arguments.get("verifyReplicaCount");
-            	if (verifyReplicaCount != null)
-            		step.setVerifyReplicaCount(verifyReplicaCount.toString());
-            }
-            
-            ParamVerify.updateDSLBaseStep(arguments, step);
-            return step;
-        }
-    }
+	public static class DescriptorImpl extends AbstractStepDescriptorImpl implements IOpenShiftTimedStepDescriptor{
+		private String wait;
+		private String waitUnit = "sec";
+
+
+		public DescriptorImpl() {
+			super(OpenShiftScalerExecution.class);
+		}
+
+		@Override
+		public String getFunctionName() {
+			return "openshiftScale";
+		}
+
+		@Override
+		public String getDisplayName() {
+			return DISPLAY_NAME;
+		}
+
+		@Override
+		public Step newInstance(Map<String, Object> arguments) throws Exception {
+			if (!arguments.containsKey("deploymentConfig") && !arguments.containsKey("depCfg"))
+				throw new IllegalArgumentException("need to specify deploymentConfig");
+			Object depCfg = arguments.get("deploymentConfig");
+			if (depCfg == null || depCfg.toString().length() == 0)
+				depCfg = arguments.get("depCfg");
+			if (depCfg == null || depCfg.toString().length() == 0)
+				throw new IllegalArgumentException("need to specify deploymentConfig");
+			if (!arguments.containsKey("replicaCount"))
+					throw new IllegalArgumentException("need to specif replicaCount");
+			OpenShiftScaler step = new OpenShiftScaler(depCfg.toString(),
+					arguments.get("replicaCount").toString());
+
+			if (arguments.containsKey("waitTime")) {
+				Object waitTime = arguments.get("waitTime");
+				if (waitTime != null) {
+					wait = waitTime.toString();
+					step.setWaitTime(wait);
+				}
+			}
+
+			if (arguments.containsKey("waitUnit")) {
```

Comment:
leverage the new default method on IOpenShiftTimedStepDescriptor called for earlier